src/controllers/users_controller.test.ts

```diff
@@ -0,0 +1,58 @@
+import supertest from 'supertest';
+
+import initApp from '../express_app';
+import {User} from '../types/model/user';
+import {DatabaseService} from '../types/database_service';
+import {Logger, newTestLogger} from '../utils/logger';
+
+describe('UsersController', () => {
+    let app;
+    let testLogger: Logger;
+    let testDatabaseService: DatabaseService;
+
+    const testUser: User = {
+        username: 'myusername',
+        first_name: 'First',
+        last_name: 'Last',
+    }
+
+    beforeEach(() => {
+        testLogger = newTestLogger();
+        testDatabaseService = {
+            users: {
+                get: jest.fn().mockResolvedValue(testUser),
+            }
+        }
+
+        app = initApp({
+            log: testLogger,
+            database: testDatabaseService,
```

Comment:
Ideally we mock something more low-level, like the actual db client, instead of mocking our own service. Then we can assert that the correct things are happening throughout our own code